homeassistant/components/alphaess/sensor.py

```diff
@@ -0,0 +1,160 @@
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.SolarToBattery,
+        name="Solar to Battery",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.SolarToGrid,
+        name="Solar to Grid",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.SolarToLoad,
+        name="Solar to Load",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.TotalLoad,
+        name="Total Load",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.GridToLoad,
+        name="Grid to Load",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.GridToBattery,
+        name="Grid to Battery",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.StateOfCharge,
+        name="State of Charge",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.Charge,
+        name="Charge",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    AlphaESSSensorDescription(
+        key=AlphaESSNames.Discharge,
+        name="Discharge",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+]
+
+
+async def async_setup_entry(hass, entry, async_add_entities) -> None:
+    """Defer sensor setup to the shared sensor module."""
+
+    coordinator: AlphaESSDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: List[AlphaESSSensor] = []
+
+    key_supported_states = {
+        description.key: description for description in SENSOR_DESCRIPTIONS
+    }
+
+    for invertor in coordinator.data:
+        serial = invertor
```

Comment:
Why is this required?

Reply:
This is the same situation as I note in https://github.com/home-assistant/core/pull/62801#discussion_r814142234  basically AlphaESS supports someone owning multiple inverters accessed under a single login to their API/website.  I want to make sure that I query all of the inverters where there are multiples.  Is there a better more home assistant best practice way of doing this?

Reply:
In this case I meant why do you need to assign invertor to serial directly. Just say `for serial in coordinator.data` otherwise.